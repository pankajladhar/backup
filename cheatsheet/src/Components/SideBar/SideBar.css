.SideBar {
    
}
.SideBar__Logo {
    margin: 0;
    font-weight: normal;
    padding: 16px;
    font-size: 20px;
    line-height: 1;
    border-bottom: 1px #e8e8e8 solid;
}

.SideBar__List {
    padding: 16px;
}

.SideBar__List a {
    line-height: 1;
    font-size: inherit;
    color: #1978c8;
}

.SideBar__List a:hover  {
    color: #f28a25;
}
