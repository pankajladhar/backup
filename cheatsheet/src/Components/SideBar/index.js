import React from 'react';
import PropTypes from 'prop-types';
import Logo from './../Logo';
import './SideBar.css';

const SideBar = (props) => {
    return(
        <aside className="SideBar">
            <Logo />
            <h2 className="SideBar__Logo">{props.heading}</h2>
            <ul className="SideBar__List">
                {
                    props.options.map((option)=>{
                        return(
                            <li key={option.id}>
                                <a href={`/cheatsheet/markdown/#${option.id}`}>{option.title}</a>
                            </li>
                        )
                    })
                }
            </ul>
        </aside>
    )
}

SideBar.propTypes = {
    heading: PropTypes.string,
    options: PropTypes.arrayOf(
        PropTypes.shape({
            options: PropTypes.string,
            title: PropTypes.string,
        })
    )
}

export default SideBar;