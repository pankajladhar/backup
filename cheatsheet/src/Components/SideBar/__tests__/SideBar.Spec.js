import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom'
import SideBar from './../index';

jest.mock('./../../Logo', () => 'Logo')

describe('SideBar', () => {
    it('renders correctly', () => {
        const sideBarProps = {
            heading: 'Sidebar',
            options: [
                {
                    "id": "id1",
                    "title": "title1",
                },
                {
                    "id": "id2",
                    "title": "title2",
                }
            ]
        }
        let tree = renderer.create(
            <StaticRouter location="someLocation" context={{}}>
                <SideBar {...sideBarProps} />
            </StaticRouter>
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
});