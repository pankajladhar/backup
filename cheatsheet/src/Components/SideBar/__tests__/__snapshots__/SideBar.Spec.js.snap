// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly 1`] = `
<aside
  className="SideBar"
>
  <a
    className="SideBar__MainLogo"
    href="/cheatsheet"
    onClick={[Function]}
  >
    <span
      className="sr-only"
    >
      Cheatsheet
    </span>
  </a>
  <h2
    className="SideBar__Logo"
  >
    Sidebar
  </h2>
  <ul
    className="SideBar__List"
  >
    <li>
      <a
        href="/cheatsheet/markdown/#id1"
      >
        title1
      </a>
    </li>
    <li>
      <a
        href="/cheatsheet/markdown/#id2"
      >
        title2
      </a>
    </li>
  </ul>
</aside>
`;
